sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -691,7 +691,7 @@ void postFlush() {
           @Nullable
           TableSchema updatedTableSchema =
```

Comment:
wait, we actually can only request for updatedTableSchema if autoUpdateSchema, that is include 
```
          TableSchema updatedTableSchema =
              (streamAppendClient != null) ? streamAppendClient.getUpdatedSchema() : null;
          if (updatedTableSchema != null) {
            invalidateWriteStream();
            appendClientInfo =
                Preconditions.checkStateNotNull(getAppendClientInfo(false, updatedTableSchema));
          }
```
in an if block.

Reply:
Thanks!

This would make it clearer, but not needed here for now -- `streamAppendClient.getUpdatedSchema()` does not make external calls, it's pulling a reference that already exists, so there's no practical difference for changing this.

We'll be addressing the bug when the update has to actually happen as part of another PR, and we can consider this path there. WDYT?